### ISSUE TITLE
Add support for JMX authentication

### ISSUE THREAD

**Issue by vbatoufflet** [zegelin/cassandra-exporter#1]
Hi,

I did dig a little into the exporter code and it seems that's not possible to handle authentication with JMX.

When I try to connect to a remote JMX instance requiring authentication I get:

```
Exception in thread "main" picocli.CommandLine$ExecutionException: Error while calling command (com.zegelin.prometheus.cassandra.Application@3ada9e37): java.lang.SecurityException: Authentication failed! Credentials required
        at picocli.CommandLine.execute(CommandLine.java:924)
        at picocli.CommandLine.access$700(CommandLine.java:104)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:1081)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:1049)
        at picocli.CommandLine$AbstractParseResultHandler.handleParseResult(CommandLine.java:957)
        at picocli.CommandLine.parseWithHandlers(CommandLine.java:1240)
        at picocli.CommandLine.call(CommandLine.java:1511)
        at picocli.CommandLine.call(CommandLine.java:1435)
        at com.zegelin.prometheus.cassandra.Application.main(Application.java:49)
Caused by: java.lang.SecurityException: Authentication failed! Credentials required
        at com.sun.jmx.remote.security.JMXPluggableAuthenticator.authenticationFailure(JMXPluggableAuthenticator.java:211)
        at com.sun.jmx.remote.security.JMXPluggableAuthenticator.authenticate(JMXPluggableAuthenticator.java:163)
        at sun.management.jmxremote.ConnectorBootstrap$AccessFileCheckerAuthenticator.authenticate(ConnectorBootstrap.java:219)
        at javax.management.remote.rmi.RMIServerImpl.doNewClient(RMIServerImpl.java:232)
        at javax.management.remote.rmi.RMIServerImpl.newClient(RMIServerImpl.java:199)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:497)
        at sun.rmi.server.UnicastServerRef.dispatch(UnicastServerRef.java:323)
        at sun.rmi.transport.Transport$1.run(Transport.java:200)
        at sun.rmi.transport.Transport$1.run(Transport.java:197)
        at java.security.AccessController.doPrivileged(Native Method)
        at sun.rmi.transport.Transport.serviceCall(Transport.java:196)
        at sun.rmi.transport.tcp.TCPTransport.handleMessages(TCPTransport.java:568)
        at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run0(TCPTransport.java:826)
        at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.lambda$run$241(TCPTransport.java:683)
        at java.security.AccessController.doPrivileged(Native Method)
        at sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run(TCPTransport.java:682)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
        at sun.rmi.transport.StreamRemoteCall.exceptionReceivedFromServer(StreamRemoteCall.java:276)
        at sun.rmi.transport.StreamRemoteCall.executeCall(StreamRemoteCall.java:253)
        at sun.rmi.server.UnicastRef.invoke(UnicastRef.java:162)
        at javax.management.remote.rmi.RMIServerImpl_Stub.newClient(Unknown Source)
        at javax.management.remote.rmi.RMIConnector.getConnection(RMIConnector.java:2430)
        at javax.management.remote.rmi.RMIConnector.connect(RMIConnector.java:308)
        at javax.management.remote.JMXConnectorFactory.connect(JMXConnectorFactory.java:270)
        at com.zegelin.prometheus.cassandra.Application.call(Application.java:38)
        at com.zegelin.prometheus.cassandra.Application.call(Application.java:20)
        at picocli.CommandLine.execute(CommandLine.java:917)
        ... 8 more
```

Do you think this feature could be added to this exporter?

Best regards,
Vincent

**Comment by zegelin** [comment]
Thanks for the report and for the PR. Looks good.

**Comment by zegelin** [comment]
Released as [v0.9.1](https://github.com/zegelin/cassandra-exporter/releases/tag/v0.9.1)

**Comment by vbatoufflet** [comment]
Great. Thanks for the quick release!
